clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -460,6 +485,7 @@ private static int parseAcks(String acksString) {
             // first make sure the metadata for the topic is available
             ClusterAndWaitTime clusterAndWaitTime = waitOnMetadata(record.topic(), record.partition(), maxBlockTimeMs);
             long remainingWaitMs = Math.max(0, maxBlockTimeMs - clusterAndWaitTime.waitedOnMetadataMs);
+            remainingWaitMs = maybeWaitOnPid(remainingWaitMs);
```

Comment:
Discussed offline. We agreed it's not actually necessary to block here since the sequence number is not assigned until the batch is ready to be sent anyway.